src/Type/Schema.php

```diff
@@ -339,15 +341,22 @@ private function loadType(string $typeName) : ?Type
 
         $type = $typeLoader($typeName);
 
+        if (is_callable($type)) {
+            $type = self::resolveType($type);
+        }
+
         if (! $type instanceof Type) {
```

Comment:
We can never reach this point if we go through the type-safe `resolveType()` before. I think we can remove this typecheck, it seems to be a leftover from `graphql-js`. We can just let PHP do the type checking for us.

Reply:
>  if we go through the type-safe resolveType()

You don't go through it if the type is not callable.

Reply:
https://github.com/webonyx/graphql-php/pull/557/files/72ccaa301778869cafbd35d2ce6746b21872b4cf#r439629614

Reply:
?